VirtoCommerce.SitemapsModule.Data/Services/SitemapItemRecordProviders/StaticContentSitemapItemRecordProvider.cs

```diff
@@ -86,16 +88,25 @@ public virtual void LoadSitemapItemRecords(Store store, Sitemap sitemap, string
                     using (var stream = storageProvider.OpenRead(url))
                     {
                         var content = stream.ReadToString();
-                        var yamlHeader = ReadYamlHeader(content);
-                        IEnumerable<string> permalinks;
-                        yamlHeader.TryGetValue("permalink", out permalinks);
-                        var frontMatterPermalink = new FrontMatterPermalink(url.Replace(".md", ""));
-                        if (permalinks != null)
+                        var frontMatterPermalink = new FrontMatterPermalink("/");
+                        if (content.TryParseJson(out JToken token))
                         {
-                            frontMatterPermalink = new FrontMatterPermalink(permalinks.FirstOrDefault());
+                            if (token.HasValues && token.First["permalink"] != null)
```

Comment:
If token doesn't contain the `permalink` property there will be null reference reception in  the 109  line `frontMatterPermalink.ToUrl())`

Reply:
No, this variable has a value already, that assigning on line 91.

Reply:
Ah, excuse me, I missed it.